### PR TITLE
Permanent saving of some types

### PR DESCRIPTION
**Pull request by granit1986**
Several types such as `DateTime`, `byte[]` which are different from redis types, were always detected as modified and saved

### CODE REVIEW

**Comment by granit1986** [comment]
@slorello89 I don`t understand what it's happened. Local this tests works fine

**Comment by slorello89** [comment]
@granit1986 - it looks like you are encountering an issue with timezones, I don't think there's anything wrong with thow the code is operating at least insofar as `DateTime` is concerned. A `DateTime` is converted to a `DateTimeOffset` with the local timezone information so it can be converted back and forth to the same `DateTime` as the server that constructed it was using.

I'm able to revert your changes and still have those tests pass so.

**Comment by granit1986** [comment]
@slorello89 

Great. Thank you. I think we can leave it like this.

**Comment by granit1986** [comment]
I was hoping that these edits would make it into the main repository.

**Comment by slorello89** [comment]
@granit1986 - I understand, but your tests pass against the original code base, so it's unclear what's being fixed by them, it sounds more like you were having a timezone issue with your DateTimes then an actual defect in Redis OM.

**Comment by granit1986** [comment]
I understand. I will try fix test for new case

File: test/Redis.OM.Unit.Tests/RediSearchTests/SearchTests.cs [hunk 1]
```diff
@@ -3840,6 +3845,67 @@ public void TestBasicQueryCastToIQueryable()
...
+
+            var dateTime = new DateTime(2024, 01, 01);
+
+            var collection = new RedisCollection<Person>(_substitute);
+            var item = collection.First();
+            item.Date = dateTime;
+            collection.Update(item);
+            
+            var timestamp = new DateTimeOffset(dateTime).ToUnixTimeMilliseconds();
+            _substitute.Received(0).Execute(
```
Comment (granit1986) [inline test/Redis.OM.Unit.Tests/RediSearchTests/SearchTests.cs:3862]: @slorello89 i changed test.
here it is implied that the date update should not occur if the value has not changed

**Pull request closed without merging**